### PR TITLE
[REF] oulook: enforce guidelines with prettier and refactor the structure of the project

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
Enforce guidelines on the project with prettier, and refactor most components.

Specification
==========
Prettier
---------
A code formatter has been configured, but never used. So, currently, the code has no real guidelines and some part of the code are very hard to read (very long line, indentation issues...).

To solve this, we configured properly prettier and run it on the code (this increase a lot the diff of this PR but the code is way cleaner like this).

Refactoring
---------------
Refactor most of the components. Some components have a lot of redundancy (`TaskListItemn`, `LeadListItem`...) and we should in that case make a common widget. Or sometimes we try to make a common component when it's counterproductive (`CompanyCard` and `ProfileCard`).

We also use new Typescript syntax (arrow functions, async / await...) as it help for readability. We remove useless state / properties, remove Typescript when it can be done in pure CSS.

Fix
---
Fix some bugs related to the enrichment (an error was returned by Odoo and never displayed on the add-on, so the user didn't know why the enrichment failed) or the "Create and Enrich" button should not be visible if the contact doesn't exist in the Odoo database.

Task-2601837

### CODE REVIEW

File: outlook/src/classes/Address.ts [hunk 1]
```diff
@@ -14,37 +14,31 @@ class Address {
...
 
-        let lines = [];
-        if (firstLine)
-            lines.push(firstLine)
-        if (secondLine)
-            lines.push(secondLine)
-        if (this.country)
-            lines.push(this.country)
+        const lines = [];
+        if (firstLine && firstLine.length) lines.push(firstLine);
```
Comment (tde-banana-odoo) [inline outlook/src/classes/Address.ts:21]: Is it standard to not use parenthesizes ? Seems you are not shocked so fine for me :) . Just noting it once.

File: outlook/src/classes/Company.ts [hunk 2]
```diff
@@ -93,89 +90,87 @@ class Company {
...
         // TODO 'sector' ?
-        const industries = [this.additionalInfo['industry_group'], this.additionalInfo['sub_industry']]
-        return industries.filter(Boolean).join(", ");
+        const industries = [this.additionalInfo['industry_group'], this.additionalInfo['sub_industry']];
+        return industries.filter(Boolean).join(', ');
     }
 
-    getEmployees() : number {
+    getEmployees(): number {
         // TODO remove the decimal
```
Comment (tde-banana-odoo) [inline outlook/src/classes/Company.ts:117]: While we are at it, it this todo still relevant ?

File: outlook/src/taskpane/components/CollapseSection/CollapseSection.tsx [hunk 3]
```diff
@@ -1,99 +1,64 @@
...
-                            {addButton}
-                            {collapseButton}
-                        </div>
-                    </div>
-                );
-            content = this.props.children;
-        }
+    render() {
+        const addButton = this.props.hasAddButton && (
+            <FontAwesomeIcon icon={faPlus} className="collapse-section-button" onClick={this.props.onAddButtonClick} />
```
Comment (tde-banana-odoo) [inline outlook/src/taskpane/components/CollapseSection/CollapseSection.tsx:38]: Could be multilined to match the styling you introduce in this PR :)

File: outlook/src/taskpane/components/App.tsx [hunk 4]
```diff
@@ -93,150 +95,196 @@ export default class App extends React.Component<AppProps, AppState> {
...
+                    <AppContext.Provider value={this.state}>
+                        <Login />
+                    </AppContext.Provider>
+                );
+            case Page.Main:
+            default:
+                return (
+                    <AppContext.Provider value={this.state}>
+                        <div
+                            style={{
```
Comment (tde-banana-odoo) [inline outlook/src/taskpane/components/App.tsx:275]: This inline style can't be removed ?

File: outlook/src/taskpane/components/Contact/ContactList/ContactListItem/ContactListItem.tsx [hunk 5]
```diff
@@ -1,71 +1,81 @@
...
-            <div onClick={this.onPartnerClick} className={classNames}>
-                {addButton}
-                <div className='contact-profile-card-container'>
-                    <ProfileCard {...profileCardData} />
+        return (
+            <div className={`contact-container ${this.props.onItemClick && 'clickable'}`} onClick={this.onPartnerClick}>
+                <div className="contact-card">
+                    {this.getIconOrInitials()}
+                    <div className="contact-info">
+                        <div className="contact-name">{jobTitle ? name + ', ' + jobTitle : name}</div>
```
Comment (tde-banana-odoo) [inline outlook/src/taskpane/components/Contact/ContactList/ContactListItem/ContactListItem.tsx:70]: Can't you build the contact_name previously ? That way we have the logic before the final rendering (like emailDiv, phoneDiv).

**Review by tde-banana-odoo — COMMENTED** [review]
Quick going through, nice cleaning and linting :) some questions arose when browsing the diff, probably some of them to discard (notably about css: sometimes css is hardcoded, I suppose there is a good reason to do so ?).

Cheers !

**Review by std-odoo — COMMENTED** [review]
@tde-banana-odoo Hi :) 

Changes done and answered :)

**Pull request merged**